lib/ws-kernel.coffee

```diff
@@ -0,0 +1,84 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+uuid = require 'uuid'
+services = require('./jupyter-js-services-shim')
+
+Kernel = require './kernel'
+
+module.exports =
+class WSKernel extends Kernel
+    constructor: (kernelSpec, @session) ->
+        super kernelSpec
+
+        @session.statusChanged.connect => @_onStatusChange()
+        @_onStatusChange() # Set initial status correctly
+
+    interrupt: ->
+        @session.kernel.interrupt()
+
+    restart: ->
+        @session.kernel.restart()
+
+    _onStatusChange: ->
+        @statusView.setStatus @session.status
+
+    _onIOPub: (message, onResults) ->
+        msg_type = message.header.msg_type
+
+        # TODO(nikita): implement support for watches
+
+        if onResults?
+            console.log('got msg')
+            console.log(message)
+            result = @_parseIOMessage(message)
+            if result?
+                onResults result
+
+
+    execute: (code, onResults) ->
+        future = @session.kernel.execute(
+            code: code
+        )
+
+        future.onIOPub = (message) =>
+            @_onIOPub(message, onResults)
+
+        future.onReply = (message) ->
+            if message.content.status is 'error'
+                return
+            result =
+                data: 'ok'
+                type: 'text'
+                stream: 'status'
+            onResults(result)
+
+    executeWatch: (code, onResults) ->
+        # TODO(nikita): implement watches
+        # Also consider switching to user_expressions for watches
+        console.log 'WARNING: Watches are not supported by remote kernels'
```

Comment:
I see about user_expressions

In terms of atom notifications, this isn't the right place (IIRC this method isn't even called anywhere, because watches are also disabled in several other places, including the one which causes executeWatch would be called). I'll figure out the entire business with watches in a future PR.